packages/core/src/availability/policyHelpers.ts

```diff
@@ -162,31 +163,26 @@ export function selectModelForAvailability(
  */
 export function applyModelSelection(
   config: Config,
-  requestedModel: string,
-  currentConfig?: GenerateContentConfig,
-  overrideScope?: string,
+  modelConfigKey: ModelConfigKey,
   options: { consumeAttempt?: boolean } = {},
-): { model: string; config?: GenerateContentConfig; maxAttempts?: number } {
-  const selection = selectModelForAvailability(config, requestedModel);
+): { model: string; config: GenerateContentConfig; maxAttempts?: number } {
+  const resolved = config.modelConfigService.getResolvedConfig(modelConfigKey);
+  const { model } = resolved;
```

Comment:
nit: use resolved.model its easier to read. I had to look up what `const { model } = resolved;` was doing